beacon-chain/p2p/discovery.go

```diff
@@ -36,24 +39,50 @@ type Listener interface {
 // to be dynamically discoverable by others given our tracked committee ids.
 func (s *Service) RefreshENR() {
 	// return early if discv5 isnt running
-	if s.dv5Listener == nil {
+	if s.dv5Listener == nil || !s.isInitialized() {
```

Comment:
dv5Listener is a nil check. Should we be using as IsNil() helper or does isInitialized take care of this?

Reply:
We do not have `IsNil` for `dv5Listener` here, it was meant to actually represent the discoveryV5 service from geth which is why we don't have additional methods. `isInitialized` has more to do with genesis, so they would technically be 'different' events.